### DISCUSSION TITLE
Version 1.5.4, the issues, and what I've learnt

### DISCUSSION THREAD

**Discussion by technosf** [louis77/tuner#150] [General]
I have been working on the Tuner code base for a month or two, after a vague promise to @louis77  a couple of years ago, in a ticket, to help when I had time. I got frustrated with Flathub being stuck at **1.4.2** while the mainline was a whole minor version ahead. I am a neophyte at Flatpaks, Meson and the new-to-me Vala language, but I recognize all the problems from every other build system, language and environment: I asked Louis to refresh Flathub and **1.5.2** went out, and I started looking at the code.

Making an impact on the functionality of the app not being a goal, I needed to understand how Tuner is built and deployed, and that meant some minor changes, documenting what I found and learning the process, if there is a process. Running Tuner from the CLI was somewhat noisy, with some orphaned tasks causing complaints when they were left roaming the threads, giving me an initial goal of getting rid of the warnings I was seeing. That was done in **v.1.5.4** (notice the '_v_' in front of the version - in process I am working to apply the standards I am hitting as I move across Tuner's touch points. Learning the Flathub system took long enough that **v1.5.3**, did not get rolled out, and it self was an initial foray into updating a library and some doc.

Release _v1.5.4_ has some issues though - namely issue #142 - relating to [Gtk GIO Settings](https://docs.gtk.org/gio/class.Settings.html) and delivering its schema in the build. I have not been able to solve this one so far, and I have not been able to tell when it became a problem: For older installs of schema, it is likely the schema is in place and _v1.5.4_ works fine. For newer installs, Tuner fails to start as GIO complains. I do not yet know if this is an issue with Meson, or a failure to packaged assets with the flatpak, and I'd be appreciative if people in the know could point the way here. I am taking the tack of knocking out a simple project with a schema in an attempt at cracking this nut.

**Comment by bztd** [comment]
I have seen the problem in other applications that use Vala related to the application of the theme. Unfortunately the only thing I know about Vala is what I have learned by reviewing the code.
